libs/platform/schematics/ng-add/index.ts

```diff
@@ -0,0 +1,229 @@
+import { Rule, SchematicContext, Tree, chain, externalSchematic } from '@angular-devkit/schematics';
+import { NodePackageInstallTask, RunSchematicTask } from '@angular-devkit/schematics/tasks';
+import { hasPackage, getSourceTreePath } from '../utils/package-utils';
+import { addPackageJsonDependency, NodeDependency, NodeDependencyType } from '@schematics/angular/utility/dependencies';
+import {
+    writeToAngularConfig,
+    createExtractionFiles,
+    replaceLibPaths,
+    addLocalizeLib
+} from '../utils/translation-utils';
+
+/**
+ * ng add schematic that
+ * - adds Core and Platform lib and their dependencies to package.json
+ * - adds `ng add @fundamental-ngx/core` external schematic to task list
+ * - adds `@angular/localize` to package.json
+ * - adds `ng add @angular/localize` external schematic to task list
+ * - adds lib translations and angular.json locale configurations to host app if no translations available
+ * - updates lib translations to host app's translations if available by appending at the end of host app's files
+ * - replaces lib source paths with node_modules source paths for lib-related trans units
+ * - installs dependent libraries and runs their external schematics
+ * @param _options options passed for this schematic
+ */
+export function ngAdd(_options: any): Rule {
+    return (_tree: Tree, _context: SchematicContext) => {
+        return chain([
+            addCoreLib(_options),
+            callCoreSchematic(_options),
+            addLocalizeLib(_options),
+            readTranslationFiles(_options),
+            endInstallTask()
```

Comment:
Let's make this install task optional, There are cases where you want to run npm i, yarn i on your own.  Just add additional question to your JSON and schema.

Reply:
should we make this optional only for core lib, or for all dependent libraries added to package.json by this schematic?

Reply:
This is only for platform.

Reply:
Sorry, I might have stated it incorrectly. What I meant was platform in turn calls core libs and also calls localize lib, so do we want to make all of these libs optional to be installed?

Reply:
We could right..

Reply:
yeah, no harm.. will do it

Reply:
done(for ng add)
also, looks like there is some problem with the version of @angular/schematics. When running core schematics against Ang10 in a different app, it throws the error:
`Cannot find module '@schematics/angular/utility/workspace'`
![image](https://user-images.githubusercontent.com/53509521/88404439-86c09c80-cdeb-11ea-9c2e-9d9e9e8b1dd8.png)
It runs the localize schematic fine, but fails the core schematic.

Reply:
I think there is already opened issue for this

Reply:
The issue https://github.com/SAP/fundamental-ngx/issues/2835 was closed with the Ang10 migration PR. I have dropped a comment highlighting this issue which occurs only when host application is upgraded from a previous version to Ang10.

Reply:
@JKMarkowski has responded on https://github.com/SAP/fundamental-ngx/issues/2835 the following: 

> Hi @kavya-b It's because core@0.20 doesn't have angular 10 support. It will be included in 0.21.0

@fkolar Ignoring the above issue for this version as this should get fixed automatically in the next ngx version.